tests/prague/eip7692_eof_v1/eip7069_extcall/test_address_space_extension.py

```diff
@@ -0,0 +1,227 @@
+            call_suffix = [0, 0, 0, 0]
+            ase_ready_opcode = False
+        case Op.EXTCALL:
+            call_suffix = [0, 0, 0]
+            ase_ready_opcode = True
+        case Op.EXTDELEGATECALL | Op.EXTSTATICCALL:
+            call_suffix = [0, 0]
+            ase_ready_opcode = True
+        case _:
+            raise ValueError("Unexpected opcode ", target_opcode)
+
+    pre = {
+        TestAddress: Account(
+            balance=1000000000000000000000,
+            nonce=1,
+        ),
+        address_entry_point: Account(
+            code=(
+                Op.MSTORE(0, Op.PUSH32(target_address))
+                + Op.SSTORE(
+                    slot_top_level_call_status,
+                    Op.CALL(50000, address_caller, 0, 0, 32, 0, 0),
+                )
+                + Op.STOP()
+            ),
+            nonce=1,
+        ),
+        address_caller: Account(
+            code=Container(
+                sections=[
+                    Section.Code(
+                        code=Op.SSTORE(
+                            slot_target_call_status,
+                            target_opcode(Op.CALLDATALOAD(0), *call_suffix),
+                        )
+                        + Op.RETURNDATACOPY(0, 0, Op.RETURNDATASIZE)
+                        + Op.SSTORE(slot_target_returndata, Op.MLOAD(0))
+                        + Op.STOP,
+                        code_inputs=0,
+                        code_outputs=NON_RETURNING_SECTION,
+                        max_stack_height=1 + len(call_suffix),
+                    )
+                ],
+            )
+            if ase_ready_opcode
+            else Op.SSTORE(
+                slot_target_call_status,
+                target_opcode(Op.GAS, Op.CALLDATALOAD(0), *call_suffix),
+            )
+            + Op.RETURNDATACOPY(0, 0, Op.RETURNDATASIZE)
+            + Op.SSTORE(slot_target_returndata, Op.MLOAD(0))
+            + Op.STOP,
+            nonce=1,
+        ),
+    }
+    match target_account_type:
+        case "empty":
+            # add no account
+            pass
+        case "EOA":
+            pre[Address(stripped_address)] = Account(code="", balance=10**18, nonce=9)
+        case "LegacyContract":
+            pre[Address(stripped_address)] = Account(
+                code=Op.MSTORE(0, Op.ADDRESS) + Op.RETURN(0, 32),
+                balance=0,
+                nonce=0,
+            )
+        case "EOFContract":
+            pre[Address(stripped_address)] = Account(
+                code=Container(
+                    sections=[
+                        Section.Code(
+                            code=Op.MSTORE(0, Op.ADDRESS) + Op.RETURN(0, 32),
+                            code_outputs=NON_RETURNING_SECTION,
+                            max_stack_height=2,
+                        )
+                    ],
+                ),
+                balance=0,
+                nonce=0,
+            )
+
+    target_storage: dict[int, int | bytes | Address] = {}
```

Comment:
I think this variable should be called `caller_storage` because the actual target's storage is not being modified.